File: backend/springboot/src/main/java/com/jobplatform/controller/AIController.java
package com.jobplatform.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/ai")
@RequiredArgsConstructor
@CrossOrigin(origins = "*")
public class AIController {
    
    @PostMapping("/generate-description")
    public ResponseEntity<?> generateDescription(@RequestBody Map<String, Object> request) {
        String jobTitle = (String) request.get("job_title");
        String companyName = (String) request.get("company_name");
        List<String> keySkills = (List<String>) request.get("key_skills");
        
        if (jobTitle == null || jobTitle.isEmpty()) {
            return ResponseEntity.badRequest().body(Map.of("error", "Job title is required"));
        }
        
        // Mock AI-generated description (replace with actual OpenAI API call in production)
        String skillsText = (keySkills != null && !keySkills.isEmpty()) 
            ? " with expertise in " + String.join(", ", keySkills) 
            : "";
        
        String description = String.format(
            "We are seeking a talented %s%s to join our team%s.\n\n" +
            "Responsibilities:\n" +
            "• Lead and execute key projects\n" +
            "• Collaborate with cross-functional teams\n" +
            "• Drive innovation and continuous improvement\n" +
            "• Mentor junior team members\n\n" +
            "Qualifications:\n" +
            "• 3+ years of relevant experience\n" +
            "• Strong problem-solving skills\n" +
            "• Excellent communication abilities\n" +
            "• Bachelor's degree in relevant field\n\n" +
            "Benefits:\n" +
            "• Competitive salary\n" +
            "• Health insurance\n" +
            "• Flexible working hours\n" +
            "• Professional development opportunities",
            jobTitle,
            skillsText,
            companyName != null ? " at " + companyName : ""
        );
        
        return ResponseEntity.ok(Map.of("description", description));
    }
}

File: backend/laravel/app/Http/Controllers/AIController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;

class AIController extends Controller
{
    public function generateDescription(Request $request)
    {
        $request->validate([
            'job_title' => 'required|string',
            'company_name' => 'nullable|string',
            'key_skills' => 'nullable|array',
        ]);

        $jobTitle = $request->job_title;
        $companyName = $request->company_name;
        $keySkills = $request->key_skills ?? [];

        // Mock AI-generated description (replace with actual OpenAI API call in production)
        // To use real OpenAI API, install openai-php/client and configure API key
        /*
        $client = \OpenAI::client(env('OPENAI_API_KEY'));
        $prompt = "Generate a professional job description for {$jobTitle}" .
                  ($companyName ? " at {$companyName}" : "") . ". ";
        if (!empty($keySkills)) {
            $prompt .= "Required skills: " . implode(', ', $keySkills) . ". ";
        }
        $prompt .= "Include responsibilities, qualifications, and benefits.";
        
        $result = $client->chat()->create([
            'model' => 'gpt-3.5-turbo',
            'messages' => [
                ['role' => 'user', 'content' => $prompt],
            ],
        ]);
        $description = $result->choices[0]->message->content;
        */

        // Mock response
        $skillsText = !empty($keySkills) ? " with expertise in " . implode(', ', $keySkills) : "";
        $description = "We are seeking a talented {$jobTitle}{$skillsText} to join our team" .
                       ($companyName ? " at {$companyName}" : "") . ".

Responsibilities:
• Lead and execute key projects
• Collaborate with cross-functional teams
• Drive innovation and continuous improvement
• Mentor junior team members

Qualifications:
• 3+ years of relevant experience
• Strong problem-solving skills
• Excellent communication abilities
• Bachelor's degree in relevant field

Benefits:
• Competitive salary
• Health insurance
• Flexible working hours
• Professional development opportunities";

        return response()->json(['description' => $description]);
    }
}
